eventsourcing/types.py

```diff
@@ -0,0 +1,236 @@
+from abc import ABC, ABCMeta, abstractmethod
+from typing import Generic, List, Optional, TypeVar, Union
+from uuid import UUID
+
+# Need to deal with the fact that Python3.6 had GenericMeta.
+try:
+    from typing import GenericMeta
+except ImportError:
+    # In Python 3.7, we can just do this:
+    class MetaAbstractDomainEntity(ABCMeta):
+        pass
+
+
+else:
+    # But in Python3.6 we must instead do this:
+    class MetaAbstractDomainEntity(GenericMeta):
+        pass
```

Comment:
Just confirming if there's a semantic difference between the above, and something like:

```python
# Need to deal with the fact that Python3.6 had GenericMeta.
try:
    from typing import GenericMeta
    class MetaAbstractDomainEntity(GenericMeta):
        pass
except ImportError:
    # In Python 3.7, we can just do this:
    class MetaAbstractDomainEntity(ABCMeta):
        pass
```

Reply:
Yes, that looks nicer. I actually changed it to look like this, then noticed mypy was complaining about redefinition of the metaclass. So I moved the 3.7 version up, and then there's the try/except with a "type: ignore" comment. Mypy seems happy with that. Will push changes....

Have also removed more of the "ignore-errors" in setup.cfg. And fixed up some more things. Mypy is green again now...